### ISSUE TITLE
[FP] Foreach usage possible: list() = each() construct

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1222]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 2.0.13 |

Current behaviour
--
For the following code, the replacement of while(list() = each()) by foreach is suggested but the result is not the same
```php
$arr = array(1,2,3,4,5,6,7,8,9);

foreach($arr as $key=>$value)
{
    unset($arr[$key + 1]);
    echo $value . PHP_EOL;
}

echo "-----". PHP_EOL;
$arr = array(1,2,3,4,5,6,7,8,9);

while (list($key, $value) = each($arr))
{
    unset($arr[$key + 1]);
    echo $value . PHP_EOL;
}
```
https://3v4l.org/lGYXb

My understanding is that the internal pointer is modifier by each() but it's not with foreach. When unset destroy the next element of the array, the pointer move and the result is not the same.

This inspection should not be reported when the internal pointer is modified in the loop by functions like unset, next, reset...

I found this issue here: https://secure.php.net/manual/en/control-structures.foreach.php#88578

(BTW, the suggestion says "Foreach should be used instead (8x faster)." I believe "each() is deprecated since PHP7.2" would be a more useful fact for this suggestion)

**Comment by kalessil** [comment]
Fixed! +updated message text.

**Comment by orklah** [comment]
Nice, Thanks!
